File: source/WeddingManager.Web/js/services/customerService.js
(function(){
    angular.module('weddingManager')
    .service('customerService', function(customerFactory, appStateService){
        var service = {};
        var _customers = [];
        var _summary = {};
        
        service.createCustomer = function(companyId, onSuccess, onError){
            var customer = {
                FirstName : "New",
                LastName : "Customer",
                PhoneNumber : ""
            };
            customerFactory.createCustomer(companyId, customer, function(customerId){
                customer.Id = customerId;
                onSuccess(customer);
            }, onError);
        }
                
        service.refreshCustomers = function(companyId, onSuccess, onError){
            customerFactory.retrieveCustomers(companyId, function(customers){
                _customers = customers;
                onSuccess(_customers);                
            }, onError);
        }
                
        service.updateCustomer = function(customer, onSuccess, onError){
            customerFactory.updateCustomer(customer, function(){
                onSuccess();
            }, onError);
        }
        
        service.deleteCustomer = function(customerId, onSuccess, onError){
            customerFactory.deleteCustomer(customerId, function(){
                appStateService.setCustomer(null);
                onSuccess();
            }, onError);
        }
        
        service.refreshSummary = function(companyId, onSuccess, onError){
            customerFactory.retrieveSummary(companyId, function(summary){
                _summary = summary;
                onSuccess(_summary);                
            }, onError);            
        }
        
        return service;
    });
})();

File: source/WeddingManager.Services/CustomerService.cs
using System.Collections.Generic;
using WeddingManager.Core.Data;
using WeddingManager.Core.Repositories;
using WeddingManager.Core.Services;

namespace WeddingManager.Services
{
    public class CustomerService : ICustomerService
    {
        public CustomerService(ICustomerRepository customerRepository)
        {
            _customerRepository = customerRepository;
        }

        public int CreateCustomer(int companyId, Customer customer)
        {
            return _customerRepository.CreateCustomer(companyId, customer);
        }

        public IEnumerable<Customer> RetrieveCustomers(int companyId)
        {
            return _customerRepository.RetrieveCustomers(companyId);
        }

        public void UpdateCustomer(Customer customer)
        {
            _customerRepository.UpdateCustomer(customer);
        }

        public void DeleteCustomer(int customerId)
        {
            _customerRepository.DeleteCustomer(customerId);
        }

        public CustomersSummary GetSummary(int companyId)
        {
            return _customerRepository.RetrieveSummary(companyId);
        }

        private readonly ICustomerRepository _customerRepository;
    }
}